ludwig/schema/config_object.py

```diff
@@ -0,0 +1,501 @@
+import copy
+import sys
+import warnings
+from dataclasses import dataclass
+from typing import Dict, List
+
+import yaml
+from marshmallow import ValidationError
+
+from ludwig.constants import (
+    ACTIVE,
+    BINARY,
+    CATEGORY,
+    COLUMN,
+    COMBINER,
+    DECODER,
+    DEFAULT_VALIDATION_METRIC,
+    DEFAULTS,
+    ENCODER,
+    EXECUTOR,
+    HYPEROPT,
+    INPUT_FEATURES,
+    LOSS,
+    MODEL_ECD,
+    MODEL_GBM,
+    MODEL_TYPE,
+    NAME,
+    NUMBER,
+    OPTIMIZER,
+    OUTPUT_FEATURES,
+    PREPROCESSING,
+    PROC_COLUMN,
+    RAY,
+    SEQUENCE,
+    SPLIT,
+    TIED,
+    TRAINER,
+    TYPE,
+)
+from ludwig.features.feature_utils import compute_feature_hash
+from ludwig.modules.loss_modules import get_loss_cls
+from ludwig.schema import validate_config
+from ludwig.schema.combiners.base import BaseCombinerConfig
+from ludwig.schema.combiners.concat import ConcatCombinerConfig
+from ludwig.schema.combiners.utils import combiner_registry
+from ludwig.schema.decoders.utils import get_decoder_cls
+from ludwig.schema.defaults.defaults import DefaultsConfig
+from ludwig.schema.encoders.base import PassthroughEncoderConfig
+from ludwig.schema.encoders.binary_encoders import BinaryPassthroughEncoderConfig
+from ludwig.schema.encoders.utils import get_encoder_cls
+from ludwig.schema.features.utils import get_input_feature_cls, get_output_feature_cls, input_config_registry
+from ludwig.schema.optimizers import get_optimizer_cls
+from ludwig.schema.preprocessing import PreprocessingConfig
+from ludwig.schema.split import get_split_cls
+from ludwig.schema.trainer import BaseTrainerConfig, ECDTrainerConfig, GBMTrainerConfig
+from ludwig.schema.utils import BaseMarshmallowConfig, convert_submodules
+from ludwig.utils.backward_compatibility import upgrade_config_dict_to_latest_version
+from ludwig.utils.misc_utils import set_default_value
+
+DEFAULTS_MODULES = {NAME, COLUMN, PROC_COLUMN, TYPE, TIED, DEFAULT_VALIDATION_METRIC}
+
+
+class BaseFeatureContainer:
+    """Base Feature container for input and output features."""
+
+    def to_dict(self):
+        """Method for getting a dictionary representation of the input features.
+
+        Returns:
+            Dictionary of input features specified.
+        """
+        return convert_submodules(self.__dict__)
+
+    def to_list(self):
+        """Method for getting a list representation of the input features.
+
+        Returns:
+            List of input features specified.
+        """
+        return list(convert_submodules(self.__dict__).values())
+
+    def filter_features(self):
```

Comment:
nit: Add docstring.

Reply:
Will add! As a side, note I do think I actually may get rid of this though since the behavior of the config object in the SDK we've decided on is to have an ellipses truncate the long `__repr__` output instead of just selectively showing some attributes under each section since it indicates that there is more to be looked at under each respective config section.